### PR TITLE
Issue 161: by column with plot_grid

### LINKED ISSUE #161: Plotting ggplots 'by column' with plot_grid()
Currently the plot_grid() function draws ggplots by row. It would be nice if the function accepts an additional boolean parameter say 'byrow' (similar to the one in the matrix() function) and then plots ggplots either by row or by column. This would avoid users to go through an extra step of getting the plots in the desired order in the 'plotlist' argument.

Here is a link to the related discussion on stackoverflow.com:
https://stackoverflow.com/questions/63074633/draw-ggplots-in-the-plot-grid-by-column-plot-grid-function-of-the-cowplot-pa/63077508?noredirect=1#comment111656039_63077508

### PR DESCRIPTION
**Pull request by BrianLang**
closes #161
- merged to master 
- rebuilt documentation 
- added visual test 
- validated visual test

### CODE REVIEW

**Comment by BrianLang** [comment]
Somehow I'm still figuring out how to do all of this. Was there a way to do what you asked (merge to master, fix things, commit, and push) without opening a new pull request?

**Comment by clauswilke** [comment]
Yes, you can just keep pushing commits and github automatically sees them and updates the PR. No problem though, we can work with this one.

**Comment by clauswilke** [comment]
Sorry, one more issue. Can you try your plot reordering code for a case where `num_plots` is less than `rows * cols`? I think your `matrix()` call will create a warning, and you should be able to fix this by padding `1:num_plots` with `NA`s. Maybe this will work: `c(1:num_plots, rep(NA, rows * cols - num_plots))`

File: tests/testthat/test_plot_grid.R [hunk 1]
```diff
@@ -27,6 +27,10 @@ test_that("basic plot arranging works", {
   expect_doppelganger("scaling plots",
     plot_grid(p1, p2, p3, labels = "AUTO", scale = c(.9, .7, .5), nrow = 1) + theme_map() # add theme_map() for plot title
   )
+
+  expect_doppelganger("colwise arranging",
+    plot_grid(p1, NULL, p2, NULL, p3, NULL, byrow = FALSE) + theme_map()
```
Comment (clauswilke) [inline tests/testthat/test_plot_grid.R:32]: I'd delete the last `NULL` argument to test the case that the number of plots is less than `rows * cols`.

Comment (BrianLang) [reply tests/testthat/test_plot_grid.R:32]: Thanks, yes, of course this could break things. I'll check and fix it: adding a visual test that assesses the case where the number of plots is less than `rows * cols`

**Comment by BrianLang** [comment]
You were correct that having `num_plots < cols * rows` produces warnings and issues, specifically that matrix recycles inputs. So I've added your suggestion of padding with `c(1:num_plots, rep(NA, rows * cols - num_plots))` and it looks correct now.

It also is coincidental the case that the previous visual test gave a false "valid" for dropping the last NULL, so I've updated it to something that wouldn't pass given the previous implementation.
> ```
>expect_doppelganger("colwise arranging",
>    plot_grid(p1, NULL, p2, p3, NULL, byrow = FALSE) + theme_map() # add theme_map() for plot title
>  )
>```

**Comment by clauswilke** [comment]
Thanks!

**Comment by BrianLang** [comment]
Woooo! First merged pull request! Thanks for your help :-)
